api/api-cache.php

```diff
@@ -0,0 +1,132 @@
+<?php
+// these functions use $db, so scripts referencing them need to define it
+
+/**
+ * Gets the first entry from the API cache that has a matching endpoint.
+ * @param string $endpoint The endpoint for the API request
+ * @return array|null The first entry that matches or null if no entry exists
+ */
+function getFromApiCacheTable(string $endpoint)
+{
+    global $db;
+    $q = $db->query('SELECT * FROM api_cache WHERE endpoint = "' . $db->real_escape_string($endpoint) . '"');
+    if ($q) {
+        $r = $q->fetch_array(MYSQLI_ASSOC);
+        return $r;
+    }
+    return null;
+}
+
+/**
+ * Gets the first entry from the API cache that has a matching endpoint. Will not return if the entry is expired.
+ * @param string $endpoint The endpoint for the API request
+ * @return array|false|null The first entry that matches, false if the entry is expired, or null if no entry exists
+ */
+function getCached(string $endpoint)
```

Comment:
Maybe rename to `getValidCacheEntry` or `getCachedIfNotExpired` or something

Reply:
Renamed in 7cb4d48